addons/pos_coupon/static/src/js/OrderWidget.js

```diff
@@ -3,10 +3,21 @@ odoo.define('pos_coupon.OrderWidget', function (require) {
 
     const OrderWidget = require('point_of_sale.OrderWidget');
     const Registries = require('point_of_sale.Registries');
+    const { onWillRender } = owl;
 
     const PosCouponOrderWidget = (OrderWidget) =>
         class PosCouponOrderWidget extends OrderWidget {
-            getActiveProgramsProps() {
+            setup() {
+                super.setup();
+                onWillRender(() => {
+                    this.withRewardsPromoPrograms = null;
+                    this.withRewardsBookedCoupons = null;
+                    this.onNextOrderPromoPrograms = null;
+                    this.show = null;
+                    Object.assign(this, this.updateActivePrograms());
+                });
```

Comment:
```suggestion
            setup() {
                super.setup();
                this.withRewardsPromoPrograms = null;
                this.withRewardsBookedCoupons = null;
                this.onNextOrderPromoPrograms = null;
                this.show = null; 
                onWillRender(() => this.updateActivePrograms());
            }
```

And in `updateActivePrograms` you modify the attributes instead of returning

Reply:
Ok done! thank you! :)